source/time_tables/tests/departure_boards_test.cpp

```diff
@@ -771,3 +771,77 @@ BOOST_FIXTURE_TEST_CASE(base_stop_schedule_limit_per_schedule, departure_board_f
     BOOST_REQUIRE_EQUAL(sc2.date_times_size(), 1);
     BOOST_CHECK_EQUAL(sc2.date_times(0).time(), "11:30"_t);
 }
+
+BOOST_AUTO_TEST_CASE(length_of_time_test1) {
+    navitia::time_duration open(07,00,00);
+    navitia::time_duration close(23,00,00);
+    navitia::time_duration begin(00,00,00);
+    navitia::time_duration end(23,59,59);
+    BOOST_CHECK_EQUAL(length_of_time(open, close), navitia::time_duration(16,00,00));
+    BOOST_CHECK_EQUAL(length_of_time(close, open), navitia::time_duration(8,00,00));
+    BOOST_CHECK_EQUAL(length_of_time(close, close), navitia::time_duration(00,00,00));
+    BOOST_CHECK_EQUAL(length_of_time(begin, end), navitia::time_duration(23,59,59));
+    BOOST_CHECK_EQUAL(length_of_time(end, begin), navitia::time_duration(00,00,01));
+}
+
+BOOST_AUTO_TEST_CASE(between_openin_and_closing_test1) {
+    navitia::time_duration open(07,00,00);
+    navitia::time_duration close(23,00,00);
+    navitia::time_duration begin(00,00,00);
+    navitia::time_duration end(23,59,59);
+    navitia::time_duration me(16,00,00);
+    BOOST_CHECK(between_opening_and_closing(me, open, close));
+    BOOST_CHECK(!between_opening_and_closing(me, close, open));
+    BOOST_CHECK(between_opening_and_closing(open, open, close));
+    BOOST_CHECK(between_opening_and_closing(close, open, close));
+    BOOST_CHECK(between_opening_and_closing(me, begin, end));
+    BOOST_CHECK(!between_opening_and_closing(me, end, begin));
+    BOOST_CHECK(!between_opening_and_closing(me, end, begin));
+}
+
+BOOST_AUTO_TEST_CASE(line_closed_test1) {
+    navitia::time_duration dur_1(00,05,00);
+    navitia::time_duration dur_2(02,00,00);
+    navitia::time_duration dur_3(01,00,00);
+    navitia::time_duration dur_4(18,00,00);
+
+    navitia::time_duration open_1(07,00,00);
+
+    navitia::time_duration close_1(23,00,00);
```

Comment:
please make the test more readable by explaining the duration in the name of the variable (for ex `dur_23h`).